### PR TITLE
Cache logs marking, filtering, tooltipping etc., according to #1596

### PR DESCRIPTION
**Pull request by rapotek**
First of all I am not marking this as a complete fix because I don't know if all the requests proposed in #1596 and on OCPL forum are completed here.

This PR contains, apart from some minor fixes, three improvements:

- Marking the current user logs on the cache logs list with blue 2px border.
- Possibility of filtering cache logs list by log type and/or author. The filter is applied on the client side only on already retrieved log entries, but when the filter is changed and there is no item matching it, the next part of entries is retrieved until some of them matches or there are no more left to retrieve.
- Showing the tooltip for cache type icon on the cache page, containing current user activities ("Found It", "Didn't found it" or "Attended" if applicable). The tooltip is activated only if there are any activities performed by current user for selected cache, otherwise a cache type name is displayed as usual.

Each improvement can be activated independently by config setting.

This is only a first attempt to introduce above improvements, I do not doubt there will be next fixes.

Tested on Firefox and Google Chrome on linux/windows/android and seems to work even on android 4.2 default browser (I found somewhere an old mobile).

Some screenshots, English version:
log_filter1

log_filter2

### CODE REVIEW

File: getLogEntries.php [hunk 1]
```diff
@@ -18,10 +20,23 @@
...
     ddd('error');
 }
-if(isset($_REQUEST['owner_id']) && $_REQUEST['owner_id'] != ''){
-    $owner_id = $_REQUEST['owner_id'];
-} else {
-    ddd('error - owner_id');
+$cache = GeoCache::fromCacheIdFactory($geocacheId);
+if ($cache == null) {
+    ddd('error - cache is not available');
+}
```
[the remark this answers is not in the document]
Comment (kojoty) [reply getLogEntries.php]: @deg-pl - ddd zawiera w sobie wlasnie chyba exit...

[the remark this answers is not in the document]
Comment (deg-pl) [reply getLogEntries.php]: @kojoty używam d() i on nie ma exit.

[the remark this answers is not in the document]
Comment (rapotek) [reply getLogEntries.php]: `ddd()` zawiera w sobie `die()`, ale ja po prostu zmodyfikowałem istniejący kod. W innych miejscach w pobliżu też jest ta funkcja używana i rzeczywiście kończy działanie.

[the remark this answers is not in the document]
Comment (deg-pl) [reply getLogEntries.php]: O proszę - znów coś nowego wiem ;)
Tym niemniej mam wrażenie, że obsługa błędów w stronę użytkownika powinna być bardziej przyjazna, a d() czy ddd() to raczej debug dla developerów.
Wiem @rapotek że tak było w oryginalnym kodzie. Tak tylko piszę i marzę ;)

**Comment by kojoty** [comment]
@deg-pl jeśli ty też się zgadzasz - pchnij merge...

**Pull request merged**